### ISSUE TITLE
Start duplicate supervisord will remove exists pidfile

### ISSUE THREAD

**Issue by coldnight** [Supervisor/supervisor#878]
```shell

$ supervisord --version
3.3.1
$ supervisord --pidfile /tmp/test.pid
$ ls /tmp/test.pid
/tmp/test.pid
$ supervisord --pidfile /tmp/test.pid
Error: Another program is already listening on a port that one of our HTTP servers is configured to use.  Shut this program down first before starting supervisord.
For help, use /usr/local/bin/supervisord -h
$ ls /tmp/test.pid
ls: /tmp/test.pid: No such file or directory
```

**Comment by mnaberez** [comment]
Duplicate of #613.
